Codigo/CustomScenarios.bas

```diff
@@ -46,7 +46,34 @@ End Type
 
 Private CustomScenarioList As New Dictionary
 Private ScenarioUpdateList() As IBaseScenario
+Private AvailableUpdateSlots As t_IndexHeap
```

Comment:
What's this? It looks overly complicated, it was better to iterate the vector using UBOUND.